src/plugins/intel_cpu/src/transformations/utils.cpp

```diff
@@ -40,5 +40,99 @@ bool has_matmul_with_compressed_weights(const std::shared_ptr<const ov::Model>&
     return false;
 }
 
+// Check specific pattern:
+// Constant
+//     |
+// Convert  Constant
+//     \    /
+//   Subtract   Constant
+//       \     /
+//       Multiply
+//          |
+//     Convert input Constant
+//        \     /   /
+//         \   /   /
+//          Gather
+bool is_gather_with_compressed_weights(const std::shared_ptr<const ov::Node>& node) {
+    if (!ov::is_type<ov::opset8::Gather>(node)) {
+        return false;
+    }
+    if (node->get_input_size() != 3) {
+        return false;
+    }
+
+    auto is_constant_with_2d = [](const ov::Output<ov::Node>& output) {
+        auto status = false;
+        std::deque<ov::Node*> nodes_to_calculate = {output.get_node()};
+
+        while ((!status) && !nodes_to_calculate.empty()) {
+            auto current_node = nodes_to_calculate.front();
+            nodes_to_calculate.pop_front();
+
+            if (current_node->get_input_size() == 0 && ov::is_type<ov::op::v0::Constant>(current_node)) {
+                auto cur_shape = current_node->get_output_shape(0);
+                if (cur_shape.size() == 2 && cur_shape[1] == 1u) {
+                    status = true;
+                }
+            } else {
+                // not a leaf - continue to search
+                for (const auto& input_value : current_node->input_values()) {
+                    const auto& input_node = input_value.get_node();
+                    nodes_to_calculate.push_front(input_node);
+                }
+            }
+        }
```

Comment:
seems to be overcomplicated, maybe we can just check the [const] or [const+convert] case (with 2d broadcastable shapes like [N, 1]) ?